Dockerfile

```diff
@@ -0,0 +1,12 @@
+FROM node:18-alpine AS build
+WORKDIR /app
+COPY angular/package*.json ./
+RUN npm ci
+COPY . .
+WORKDIR /app/angular
+RUN npm run build
```

Comment:
this is a duplicate unit of work. You build dist/ in the github actions - use it. Otherwise you double work yourself. One unit of work is in dockerfile (docker build) and second in the GHA itself. So use GHA and make simplest dockerfile based on nginx.